pkg/server/plugin/upstreamauthority/vault/vault_client.go

```diff
@@ -162,31 +179,49 @@ func (c *ClientConfig) NewAuthenticatedClient(method AuthMethod) (*Client, error
 		}
 		sec, err := client.Auth(path, body)
 		if err != nil {
-			return nil, err
+			return nil, false, err
 		}
 		if sec == nil {
-			return nil, errors.New("approle authentication response is nil")
+			return nil, false, errors.New("approle authentication response is nil")
 		}
-		if sec.Auth.Renewable {
-			c.Logger.Debug("Token will be renewed")
-			if err := renewToken(vc, sec, c.Logger); err != nil {
-				return nil, err
-			}
-		} else {
-			c.Logger.Debug("Token is not renewable")
+		ts, err = handleRenewToken(vc, sec, c.Logger)
+		if err != nil {
+			return nil, false, err
 		}
 	}
 
-	return client, nil
+	return client, isTokenReusable(ts), nil
```

Comment:
I'm wondering if the switch for the method should have a default case returning an error?

Reply:
I agree that the method returns error for default case. 
It's better.